### DISCUSSION TITLE
Self teaching CSS, but struggling with borders and text alignment on my table--advice?

### DISCUSSION THREAD

**Discussion by daniellelamb** [community/community#126285] [Programming Help] [answered]
### Body

Hi all, I am trying to learn some web development skills during my unemployment. For a bit of practice, I am trying to redesign page 238 from Jon Duckett's _HTML and CSS: Design and Build Websites._

I understand that HTML and CSS may not be technically programming languages, but I wasn't sure where else to ask my questions. Please let me know if there is a better forum for this...

Here is my repo with html and css docs: https://github.com/daniellelamb/css-selectors.git

This is my final goal below of what I am trying to recreate:

![436380223_862250635736272_2395825099301257359_n](https://github.com/community/community/assets/168598738/b2350c10-7da7-48b7-91a0-fc88f7f6d11a)

I'm close, but I am running into the following issues:

- The third column "Example" cannot be vertically aligned to the top like the other two columns. Nothing I do seems to change it. I think because the third column has both h2 and regular text, and that is complicating it.
- The solid grey line under the column headings is supposed to be unbroken between columns, but there is a gap I'd like to remove. I would also like to collapse it on top of the dotted line it shares a space with, or remove that one dotted line entirely.
- I'd like to remove the dotted line that is above the headings (Selector, Meaning, and Example)

I have no deadline or assignment--this is just a personal puzzle I'd like to solve so I can learn more nuanced techniques. If you notice anything else in my code, happy to hear!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by spenserblack** [comment]
Some things you might want to consider:

- Try looking into the [`border-collapse`](https://developer.mozilla.org/en-US/docs/Web/CSS/border-collapse) CSS property and what it can do.
- Putting the header into `<thead>` instead of `<tbody>` will make it easier to make a CSS selector that targets the header row.

Your alignment issue in the 3rd column is caused by the default margins for `<h2>`. Semantically, you don't want to use `<h1>` - `<h6>` for just styling, because it has accessibility utilities, too. For example, screen readers can use them for navigation. You might want to look into the [accessibility concerns for `<h1>` - `<h6>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#accessibility_concerns).

So I'd recommend using `<code>` instead of `<h2>` for the examples. The styling you have is actually pretty close to [Bootstrap's styling for `<code>`](https://getbootstrap.com/docs/5.3/content/reboot/#inline-code).

Hope these tips help!

**Reply by daniellelamb** [reply]
Thank you so much! I I will explore that further. I appreciate you bringing up the accessibility concerns, since I'd like to try to implement best practices from the beginning.

**Comment by davevad93** [comment, marked as answer]
Hi @daniellelamb ,

I've taken a look at your `css` code and here are a couple of tips to achieve what you are looking for, I'm gonna show you just what you have to add to your existing code:

```css
table {
    /* Add this to collapse borders */
    border-collapse: collapse; 
}

th[scope='col'] {
    /* Add this to remove the top dotted lines */
    border-top: none; 
}

td h2 {
    /* Add this to ensure that the contents of the "Example" column are vertically aligned with the contents of the other columns */
    margin-top: 0; 
}
```
That should work, If you have any other doubts don't hesitate to ask!

**Reply by daniellelamb** [reply]
Thank you so much for taking the time to give such detailed help! It worked like a charm. 

Question: the border collapse fixed my major issue, but it led to these weird 'double dots' in the table. Are you familiar with this? Picture below:

<img width="559" alt="Screenshot 2024-05-30 101622" src="https://github.com/community/community/assets/168598738/5d89d701-6f77-4bc5-bf0f-d9b7cca66c61">

I appreciated your help greatly!

**Reply by davevad93** [reply]
Hey @daniellelamb , you're welcome I'm really glad you found it helpful. About the dot problem in the table lines, yes I'm familiar with that because I've experienced something similar in the past and I'm pretty sure It's a browser rendering issue. You know, each browser behaves differently when it comes to interpreting `CSS` styles which means that there are small details that might look slightly different depending on the browser, for example a few months ago when I was developing a web page the menus looked smaller and a bit out of place in `Google Chrome` and `MS Edge` while in `Mozilla Firefox` they were fine. There's a tool used by many frontend developers called [Reset CSS](https://meyerweb.com/eric/tools/css/reset/) with the goal of reducing browser inconsistencies to help fix problems similar to this one. So try with opening your `HTML` with a different browser like `Firefox` or `Safari` and let me know how it looks!

**Comment by ebndev** [comment]
Hey @daniellelamb, congrats on beginning your coding journey! Please let us know if any of the answers above helped and if you have any other questions 😄 

[100Devs](https://www.youtube.com/watch?v=o3IIobN4xR0) is a great channel for learning web-development. The timing of the videos may seem intimidating but they have a Discord channel you can join to connect with others doing the course as well!

Also, feel to to hop over into our New to GitHub category and [introduce yourself](https://github.com/orgs/community/discussions/23204)! 👋
